### PR TITLE
Yeksi Naa Reports | Satisfaction Rate after delivery report

### PR DESCRIPTION
**Pull request by dmydlo**
Hi @czue,
This is an 8th Yeksi Naa Report, Satisfaction Rate after delivery report.

### CODE REVIEW

File: custom/yeksi_naa_reports/sqldata.py
Comment (czue) [inline custom/yeksi_naa_reports/sqldata.py]: maybe not important, but i'm curious why a numerator is required here? 

wouldn't 0/100 be a relevant data point to include in a total percentage?

File: custom/yeksi_naa_reports/sqldata.py
Comment (czue) [inline custom/yeksi_naa_reports/sqldata.py]: nit: it would be much easier to parse this if it was extracted into a function. e.g. `if self._date_is_in_range(row['real_date_repeat'])` (or whatever an appropriate name would be)

File: custom/yeksi_naa_reports/sqldata.py
Comment (czue) [inline custom/yeksi_naa_reports/sqldata.py]: this could probably be simplified / eliminated if you used a couple nested `defaultdict`s

File: custom/yeksi_naa_reports/sqldata.py
Comment (czue) [inline custom/yeksi_naa_reports/sqldata.py]: this entire loop could maybe use a comment explaining what it's doing. or maybe better be extracted into a function with a name explaining what it's doing.

File: custom/yeksi_naa_reports/sqldata.py
Comment (czue) [inline custom/yeksi_naa_reports/sqldata.py]: (same comment as other loop)

File: custom/yeksi_naa_reports/sqldata.py
Comment (czue) [inline custom/yeksi_naa_reports/sqldata.py]: same question as above - is it not ever valid to have a numerator of `0` but a real denominator?

File: custom/yeksi_naa_reports/tests/reports/test_dashboard_3.py
Comment (czue) [inline custom/yeksi_naa_reports/tests/reports/test_dashboard_3.py]: :+1:

**Review by czue — COMMENTED** [review]
most/all of my comments are not blockers, but want to hear your response on the exclusion of numerators of 0 before merging.

File: custom/yeksi_naa_reports/sqldata.py
Comment (dmydlo) [reply custom/yeksi_naa_reports/sqldata.py]: Right, it should only check denominator existence to include only relevant data.

**Comment by dmydlo** [comment]
After rebasing with the master branch I have named loops in Satisfaction Rate after a delivery report as well as I've added use of defaultdict's.
On top of that, I have renamed rows to records across all sqldata as these come from database and we are returning table rows and added usage of created helper functions for operating on dates also across the whole document.

If you have any questions I will be happy to answer.
P.S. Thanks for great bits of advice.

File: custom/yeksi_naa_reports/sqldata.py
Comment (czue) [inline custom/yeksi_naa_reports/sqldata.py]: is it ever possible that this condition would never be hit and this function would return `None`? It seems like a lot of code below would break if that ever happened, so could be good to be explicit (e.g. by throwing an exception outside the bottom of the loop if it reaches there).

**Review by czue — APPROVED** [review]
thanks for the cleanup! I found the revised version much easier to read/reason with - hopefully you did too. 

I had one last thought/question but I'll go ahead and approve/merge this and let you follow up as/if you see fit.

File: custom/yeksi_naa_reports/sqldata.py
Comment (dmydlo) [reply custom/yeksi_naa_reports/sqldata.py]: It will never hit `None` as before each use we check `self.date_in_selected_date_range`, which checks if it would hit `None`

Comment (czue) [reply custom/yeksi_naa_reports/sqldata.py]: got it, thanks

**Pull request merged**
